### PR TITLE
Update Readme for Azure Active Directory

### PR DESCRIPTION
**Pull request by martin-loetzsch**
This PR explains better how to use this project with Azure AD

## Description

This PR only updates README.md

## Motivation and Context

I've been recently using this project in collaboration with the Azure Active Directory. The documentation in the README.md didn't help at all, I was rather following [this](https://sweetcode.io/authentication-applications-azure-active-directory-2/) and [this](https://github.com/bitly/oauth2_proxy/pull/347/files). It might be that my provided documentation is not complete, but I think it's definitely better than the current version. 

## How Has This Been Tested?

I run it in production like this in two different projects.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,11 +96,19 @@ Note: The user is checked against the group members list on initial authenticati
...
+2. Pick a name and choose "Webapp / API" as application type. Use `https://internal.yourcompany.com` as Sign-on URL. Click "Create".
+3. On the "Settings" / "Properties" page of the app, pick a logo and select "Multi-tenanted" if you want to allow users from multiple organizations to access your app. Note down the application ID. Click "Save".
+4. On the "Settings" / "Required Permissions" page of the app, click on "Windows Azure Active Directory" and then on "Access the directory as the signed in user". Hit "Save" and then then on "Grant permissions" (you might need another admin to do this).
+5. On the "Settings" / "Reply URLs" page of the app, add `https://internal.yourcompanycom/oauth2/callback` for each host that you want to protect by the oauth2 proxy. Click "Save".
+6. On the "Settings" / "Keys" page of the app, add a new key and note down the value after hitting "Save".
+7. Configure the proxy with
 
-The Azure AD auth provider uses `openid` as it default scope. It uses `https://graph.windows.net` as a default protected resource. It call to `https://graph.windows.net/me` to get the email address of the user that logs in.
+```
+   --provider=azure \
```
Comment (JoelSpeed) [inline README.md]: Can we remove the trailing slash from this line and the one below for consistency

Comment (martin-loetzsch) [reply README.md]: done

File: README.md [hunk 2]
```diff
@@ -96,11 +96,19 @@ Note: The user is checked against the group members list on initial authenticati
 
 ### Azure Auth Provider
 
-1.  [Add an application](https://azure.microsoft.com/en-us/documentation/articles/active-directory-integrating-applications/) to your Azure Active Directory tenant.
-2.  On the App properties page provide the correct Sign-On URL ie `https://internal.yourcompany.com/oauth2/callback`
-3.  If applicable take note of your `TenantID` and provide it via the `--azure-tenant=<YOUR TENANT ID>` commandline option. Default the `common` tenant is used.
+1. Add an application: go to [https://portal.azure.com](https://portal.azure.com), choose "Azure Active Directory" in the left menu, select "App registrations" and then click on "New app registration".
```
Comment (JoelSpeed) [inline README.md]: Could you make the quoted menu items bold across this PR and then I'm happy to get it merged
eg:
```suggestion
1. Add an application: go to [https://portal.azure.com](https://portal.azure.com), choose **"Azure Active Directory"** in the left menu, select **"App registrations"** and then click on **"New app registration"**.
```

File: README.md [hunk 3]
```diff
@@ -96,11 +96,19 @@ Note: The user is checked against the group members list on initial authenticati
...
+2. Pick a name and choose "Webapp / API" as application type. Use `https://internal.yourcompany.com` as Sign-on URL. Click "Create".
```
Comment (JoelSpeed) [inline README.md]:
```suggestion
2. Pick a name and choose **"Webapp / API"** as application type. Use `https://internal.yourcompany.com` as Sign-on URL. Click **"Create"**.
```

File: README.md [hunk 4]
```diff
@@ -96,11 +96,19 @@ Note: The user is checked against the group members list on initial authenticati
...
+3. On the "Settings" / "Properties" page of the app, pick a logo and select "Multi-tenanted" if you want to allow users from multiple organizations to access your app. Note down the application ID. Click "Save".
```
Comment (JoelSpeed) [inline README.md]:
```suggestion
3. On the **"Settings"** / **"Properties"** page of the app, pick a logo and select **"Multi-tenanted"** if you want to allow users from multiple organizations to access your app. Note down the application ID. Click **"Save"**.
```

File: README.md [hunk 5]
```diff
@@ -96,11 +96,19 @@ Note: The user is checked against the group members list on initial authenticati
...
+4. On the "Settings" / "Required Permissions" page of the app, click on "Windows Azure Active Directory" and then on "Access the directory as the signed in user". Hit "Save" and then then on "Grant permissions" (you might need another admin to do this).
```
Comment (JoelSpeed) [inline README.md]:
```suggestion
4. On the **"Settings"** / **"Required Permissions"** page of the app, click on **"Windows Azure Active Directory"** and then on **"Access the directory as the signed in user"**. Hit **"Save"** and then then on **"Grant permissions"** (you might need another admin to do this).
```

File: README.md [hunk 6]
```diff
@@ -96,11 +96,19 @@ Note: The user is checked against the group members list on initial authenticati
...
+5. On the "Settings" / "Reply URLs" page of the app, add `https://internal.yourcompanycom/oauth2/callback` for each host that you want to protect by the oauth2 proxy. Click "Save".
```
Comment (JoelSpeed) [inline README.md]:
```suggestion
5. On the **"Settings"** / **"Reply URLs"** page of the app, add `https://internal.yourcompanycom/oauth2/callback` for each host that you want to protect by the oauth2 proxy. Click **"Save"**.
```

File: README.md [hunk 7]
```diff
@@ -96,11 +96,19 @@ Note: The user is checked against the group members list on initial authenticati
...
+6. On the "Settings" / "Keys" page of the app, add a new key and note down the value after hitting "Save".
```
Comment (JoelSpeed) [inline README.md]:
```suggestion
6. On the **"Settings"** / **"Keys"** page of the app, add a new key and note down the value after hitting **"Save"**.
```

File: README.md (hunk 6 above)
Comment (martin-loetzsch) [reply README.md]: Done. Sorry for missing this

**Comment by JoelSpeed** [comment]
Sorry @martin-loetzsch, there's a few more suggestions open, would you mind merging them all? Just for consistency in the documentation is all 😄

**Comment by martin-loetzsch** [comment]
Ok, I hope I got them all now

**Comment by martin-loetzsch** [comment]
Better?

**Pull request merged**